api/v1beta1/ironic_webhook_test.go

```diff
@@ -0,0 +1,469 @@
+			basePath: field.NewPath("spec"),
+			expectedErrs: field.Invalid(
+				field.NewPath("spec").Child("rpcTransport"),
+				"yaml",
+				errInvalidRPCTransport,
+			),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validateRPCTransport(tc.spec, tc.basePath)
+
+			if !reflect.DeepEqual(errs, tc.expectedErrs) {
+				t.Errorf("validateRPCTransport() failed:\n    expected: %v\n    got:      %v", tc.expectedErrs, errs)
+			}
+		})
+	}
+}
+
+func TestValidateConductorSpec(t *testing.T) {
+	testCases := []struct {
+		name         string
+		spec         *IronicSpecCore
+		basePath     *field.Path
+		expectedErrs field.ErrorList
+	}{
+		{
+			name:     "IronicConductors is not",
+			spec:     &IronicSpecCore{},
+			basePath: field.NewPath("spec"),
+			expectedErrs: field.ErrorList{
+				field.Required(
+					field.NewPath("spec").Child("ironicConductors"),
+					"IronicConductors must be provided",
+				),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validateConductorSpec(tc.spec, tc.basePath)
+
+			if !reflect.DeepEqual(errs, tc.expectedErrs) {
+				t.Errorf("validateConductorSpec() failed:\n    expected: %v\n    got:      %v", tc.expectedErrs, errs)
+			}
+		})
+	}
+}
+
+func TestValidateDHCPRangesOverlap(t *testing.T) {
+	testCases := []struct {
+		name         string
+		spec         *IronicSpecCore
+		basePath     *field.Path
+		expectedErrs field.ErrorList
+	}{
+		{
+			name: "No overlap",
+			spec: &IronicSpecCore{
+				IronicConductors: []IronicConductorTemplate{
+					{DHCPRanges: []DHCPRange{{Start: "192.168.1.10", End: "192.168.1.20"}}},
+				},
+				IronicInspector: IronicInspectorTemplate{
+					DHCPRanges: []DHCPRange{{Start: "192.168.2.10", End: "192.168.2.20"}},
+				},
+			},
+			basePath:     field.NewPath("spec"),
+			expectedErrs: field.ErrorList{},
+		},
+		{
+			name: "Overlap between Conductor and Inspector",
+			spec: &IronicSpecCore{
+				IronicConductors: []IronicConductorTemplate{
+					{DHCPRanges: []DHCPRange{{Start: "192.168.1.10", End: "192.168.1.20"}}},
+				},
+				IronicInspector: IronicInspectorTemplate{
+					DHCPRanges: []DHCPRange{{Start: "192.168.1.15", End: "192.168.1.25"}},
+				},
+			},
+			basePath: field.NewPath("spec"),
+			expectedErrs: field.ErrorList{
+				field.Forbidden(
+					field.NewPath("spec").Child("ironicConductors").Index(0).Child("dhcpRanges").Index(0),
+					fmt.Sprintf(
+						errForbiddenAddressOverlap,
+						"192.168.1.10->192.168.1.20",
+						field.NewPath("spec").Child("ironicInspector").Child("dhcpRanges").Index(0),
+						"192.168.1.15->192.168.1.25",
+					),
+				),
+			},
+		},
+		{
+			name: "Overlap within same Conductors",
+			spec: &IronicSpecCore{
+				IronicConductors: []IronicConductorTemplate{
+					{DHCPRanges: []DHCPRange{
+						{Start: "192.168.1.10", End: "192.168.1.20"},
+						{Start: "192.168.1.15", End: "192.168.1.25"},
```

Comment:
Uhh... We shouldn't be permitting this. :(

Reply:
It looks like the code below asserts a Forbidden is raised